### PR TITLE
Update examples

### PR DESCRIPTION
**Pull request by jameshales**
There is now a test script for each of the examples, along with documentation for running them. As opposed to the unit tests, the test scripts run the examples against actual Hadoop/Hive/MySQL.

This is based on <https://github.com/CommBank/maestro/pull/394>.

Tested on a CDH5.3.0 virtual machine set up with Vagrant, with some configuration afterwards. Specifically, you need to delete a problematic JODA time JAR:
```
sudo mv /usr/lib/hadoop-mapreduce/joda-time-1.6.jar /usr/lib/hadoop-mapreduce/joda-time-1.6.jar.bak
```
and you need to add the following properties to `/etc/hadoop/conf/mapred-site.xml`:
```
<property>
    <name>hive.metastore.uris</name>
    <value>thrift://localhost:9083</value>
  </property>
  <property>
    <name>yarn.resourcemanager.address</name>
    <value>localhost</value>
  </property>
  <property>
    <name>hive.exec.dynamic.partition.mode</name>
    <value>nonstrict</value>
  </property>
```
Other than that, follow the instructions in `maestro-example/Readme.md`.

### CODE REVIEW

File: maestro-example/src/main/scala/au/com/cba/omnia/maestro/example/CustomerSqoopImportExecution.scala [hunk 1]
```diff
@@ -49,12 +49,28 @@ case class CustomerImportConfig(config: Config) {
...
+    // Fail if there was a problem with the load
     loadSuccess        <- loadInfo.withSuccess
+    // Write out the Customers to a Hive table
     hiveCount          <- viewHive(conf.catTable, pipe)
   } yield (CustomerImportStatus(sqoopCount, loadSuccess.written, hiveCount))
 }
 
 case class CustomerImportStatus(sqoopCount: Long, loadCount: Long, hiveCount: Long)
+
+object CustomerSqoopImportExecutionJob extends MaestroJob {
```
Comment (stephanh) [inline maestro-example/src/main/scala/au/com/cba/omnia/maestro/example/CustomerSqoopImportExecution.scala]: We could probably just turn the Execution object into a `MaestroJob` without the need of this second wrapper object?

Are you happy to do that or do you want me to do that?

**Comment by stephanh** [comment]
@jameshales very nice. Thank you for that!

**Comment by vineethvarghese** [comment]
@jameshales can please merge this when you get a chance

**Comment by stephanh** [comment]
Incorporated as part of #435.

**Pull request closed without merging**